docs/guide/batch-operations.md

```diff
@@ -0,0 +1,222 @@
+
+```javascript
+modalOptions: {
+  title: 'Reset {{ type }}',
+  description: 'Are you sure you want to reset {{ count }} {{ type }}?',
+  confirmationButton: 'Yes, reset the selected content'
+}
+```
+
+The `modalOptions` object configures the confirmation modal that appears when an editor initiates a batch operation. This confirmation step helps to prevent accidental changes to possibly hundreds of pieces. If this is not included, the batch operation's `label` is used for the title, there is no description, and the standard confirmation button label is used (e.g., "Yes, continue.").
+
+With these configuration, we should immediately see a button for the "Reset" operation in the article piece manager.
+
+![The article piece manager, now with a button using the recycle symbol](/images/batch-operation-recycle-button.png)
+
+## Adding the API route
+
+Right now if we clicked that new button and confirmed to continue nothing would happen except for an error notification saying something like "Batch operation Reset failed." Since the batch operation is called `reset`, the manager is going to look for an API route at `/v1/api/article/reset` (the piece type's base API path, plus `/reset`). We need to add that route to the piece type.
+
+Batch operation route handlers will usually have a few steps in common, so we can look at those elements in the example below.
+
+<AposCodeBlock>
+  ```javascript
+  module.export = {
+    // `batchOperations` and other module settings...
+    apiRoutes(self) {
+      return {
+        post: {
+          reset(req) {
+            // Make sure there is an `_ids` array provided.
+            if (!Array.isArray(req.body._ids)) {
+              throw self.apos.error('invalid');
+            }
+
+            // Ensure that the req object and IDs are using the same locale
+            // and mode.
+            req.body._ids = req.body._ids.map(_id => {
+              return self.inferIdLocaleAndMode(req, _id);
+            });
+
+            // Run the batch operation as a "job," passing the iterator function
+            // as an argument to actually make the changes.
+            return self.apos.modules['@apostrophecms/job'].runBatch(
+              req,
+              req.body._ids,
+              resetter,
+              {
+                action: 'reset'
+              }
+            );
+
+            // The iterator function that updates each individual piece.
+            async function resetter (req, id) {
+              const piece = await self.findOneForEditing(req, { _id: id });
+
+              if (!piece) {
+                throw self.apos.error('notfound');
+              }
+
+              // 🪄 Do the work of resetting piece field values.
```

Comment:
Seems to me we could do an actual thing here for additional clarity, but this is a big incremental improvement.

Reply:
If it were simply changing `archive` to a boolean that might be alright. I'm not aware of a way to reset only fields with default values in only a few lines. I'd like to keep the focus of this on what batch operations need to work.